### ISSUE TITLE
PSC enabled in module but getting an error

### ISSUE THREAD

**Issue by axtma** [terraform-google-modules/terraform-google-sql-db#535]
### TL;DR

PSC enabled in the definition but getting an error. I am trying to create a MySQL instance with the following config : 

```
ip_configuration = {
    ipv4_enabled        = false
    require_ssl         = false
    allocated_ip_range  = null
    authorized_networks = []
    private_network     = null
    psc_config = {
      psc_enabled                   = true
      psc_allowed_consumer_projects = [var.consumer_project_id]
    }
  }
```
But getting an error as : 

```
│ Error: Error, failed to create instance test-v89: googleapi: Error 400: Invalid request: At least one of Public IP or Private IP or PSC connectivity must be enabled.., invalid
```
This is not the same when I am using the [resource definition](https://cloud.google.com/sql/docs/mysql/configure-private-service-connect#create-cloud-sql-instance-psc-enabled) - this creates the MySQL instance correctly with PSC enabled.

### Expected behavior

Should not get an error, a MySQL instance should get created with PSC enabled and return service attachment. 

### Observed behavior

Returns an error that PSC is not enabled in config. 

### Terraform Configuration

```hcl
locals {
  ip_configuration = {
    psc_config = {
      psc_enabled                   = true
      psc_allowed_consumer_projects = [var.consumer_project_id]

    }
    ipv4_enabled        = false
    require_ssl         = false
    allocated_ip_range  = null
    authorized_networks = []
    private_network     = null
  }
}

module "sql-db" {
  source           = "../../modules/cloudsql"
  name             = var.cloudsql_instance_name
  database_version = var.database_version
  zone             = var.zone
  project_id       = var.producer_project_id
  ip_configuration = local.ip_configuration

  deletion_protection = var.deletion_protection
  depends_on = [
    module.producer_vpc
  ]
}
```

### Terraform Version

```sh
source              = "github.com/terraform-google-modules/terraform-google-sql-db//modules/mysql"
akshatmawasthi-macbookpro2:3. PSC akshatmawasthi$ terraform version
Terraform v1.5.6
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.84.0
+ provider registry.terraform.io/hashicorp/google-beta v4.84.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/template v2.2.0

My sql-db version is 16.1.
```

### Additional information

For experimentation post getting this error, I also tried to define a private network. However, if I define a [private network](https://cloud.google.com/sql/docs/mysql/private-ip#set_up_private_services_access_for_your_network) it creates a SQL instance for me with a private IP and not a PSC enabled SQL instance. Because of this, when I create a forwarding rule from the reserved IP as my PSC endpoint to the PSC attachment as a target, I get : 

```
│ Error: Error creating ForwardingRule: googleapi: Error 400: Invalid value for field 'resource.target': ''. No target or backend service specified for forwarding rule., invalid
```

I can then confirm that this isn't a PSC enabled SQL instance as `gcloud sql instances describe test-ddh --project=akshat-cncs-test | grep "pscServiceAttachmentLink"` returns nothing.

**Comment by g-awmalik** [comment]
Can you share how you're setting `ip_configuration` inside your `../../modules/cloudsql` module? 

From first looks it appears that your structure for `ip_configuration` is slightly off. In the [mysql](https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/modules/mysql) module, the [`psc_enabled`](https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/modules/mysql/variables.tf#L220) attribute is directly under [`ip_configuration`](https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/modules/mysql/variables.tf#L211) vs. your local version of `ip_configuration` has it nested under `psc_config`.

Since I can't see what's going on in `../../modules/cloudsql`, my suspicion is that the right attributes i.e. `psc_enabled` & `psc_allowed_consumer_projects` aren't getting set appropriately in the mysql module hence erroring out. A [working example of mysql using PSC](https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/examples/mysql-psc) can be found here for reference.

I just provisioned an instance of mysql on PSC using that example and it worked fine. Here's what it looks like:
![image](https://github.com/terraform-google-modules/terraform-google-sql-db/assets/21246369/707efd29-306a-419a-9ead-9eb21ed3b190)

**Comment by axtma** [comment]
Hi,

Thanks for your response. Your guidance with `setting ip_configuration inside your ../../modules/cloudsql module?` was very helpful as this indeed was an issue with the module. In the variables.tf I had to set some variables as optional and add psc_enabled = optional(bool) & psc_allowed_consumer_projects = optional(list(string)).

Best
